com/win32com/client/__init__.py

```diff
@@ -487,6 +487,46 @@ def Record(name, object):
     )
 
 
+# Registration function for com_record subclasses.
+def register_record_class(cls):
+    """
+    Register a subclass of com_record to enable creation of the represented record objects.
+
+    A subclass of com_record requires the following class attributes to be instantiable:
+
+        TLBID : The GUID of the containing TypeLibrary as a string.
+        MJVER : The major version number of the TypeLibrary as an integer.
+        MNVER : The minor version number of the TypeLibrary as an integer.
+        LCID  : The LCID of the TypeLibrary as an integer.
+        GUID  : The GUID of the COM Record as a string.
+
+    with GUID strings in {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx} notation.
+
+    To instantiate such a subclasses it has to be registered via this function.
+    """
+    if cls not in pythoncom.com_record.__subclasses__():
+        raise TypeError("Only subclasses of 'com_record' can be registered.")
+    try:
+        obj = cls()
+    except:
+        raise TypeError(f"Class {cls.__name__} cannot be instantiated.")
```

Comment:
I can't speak on the c++ side. But on the Python side, it is generally good practice to be specific about the expected error. As to not accidentally silence other errors that could reveal different problems.

Alternatively, since you're re-raising, if you're gonna catch-all anyway, you may include the previous error to have a clearer trace:
```suggestion
    except Exception as e:
        raise TypeError(f"Class {cls.__name__} cannot be instantiated.") from e
```

Reply:
You're right. Thank you for the suggested enhancement.